src/interfaces/qwbufferinterface.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2023 Dingyuan Zhang <zhangdingyuan@uniontech.com>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#pragma once
+
+#include <qwinterface.h>
+
+struct wlr_buffer;
+struct wlr_buffer_impl;
+struct wlr_dmabuf_attributes;
+struct wlr_shm_attributes;
+
+class QWBuffer;
+
+QW_BEGIN_NAMESPACE
+
+class QWBufferInterface : public QWInterface
+{
+public:
+    virtual ~QWBufferInterface();
+
+    bool getDmabuf(wlr_dmabuf_attributes *attribs);
```

Comment:
you need to check the wlroots project, if the getDmabuf is a necessary interface, we should make it to the full virtual function.